django/core/checks/translation.py

```diff
@@ -1,19 +1,68 @@
 from django.conf import settings
 from django.utils.translation.trans_real import language_code_re
 
-from . import Error, Tags, register
+from . import Error, Tags, Warning, register
 
 E001 = Error(
-    'You have provided an invalid value for the LANGUAGE_CODE setting.',
+    'You have provided an invalid value for the LANGUAGE_CODE setting: {!r}.',
     id='translation.E001',
 )
 
+E002 = Error(
+    'You have provided an invalid language code in the LANGUAGES setting: {!r}.',
+    id='translation.E002',
+)
+
+E003 = Error(
+    'You have provided an invalid language code in the LANGUAGES_BIDI setting: {!r}.',
+    id='translation.E003',
+)
+
+E004 = Error(
+    'You have provided a value for the LANGUAGE_CODE setting that is not in the LANGUAGES setting.',
+    id='translation.E004',
+)
+
+W005 = Warning(
```

Comment:
I went for warning as there is no case where this would cause things to go wrong. It was just to highlight inconsistent settings. But we can make it an error to be consistent if you'd prefer.